scripts/exregional_process_bufr.sh

```diff
@@ -0,0 +1,417 @@
+#
+print_info_msg "$VERBOSE" "
+Getting into working directory for BUFR obseration process ..."
+
+cd ${WORKDIR}
+
+fixdir=$FIXgsi/
+
+print_info_msg "$VERBOSE" "fixdir is $fixdir"
+
+#
+#-----------------------------------------------------------------------
+#
+# link or copy background files
+#
+#-----------------------------------------------------------------------
+
+cp_vrfy ${fixdir}/fv3_grid_spec          fv3sar_grid_spec.nc
+cp_vrfy ${fixdir}/geo_em.d01.nc          geo_em.d01.nc
+
+
+#-----------------------------------------------------------------------
+#
+#   copy bufr table
+#
+#-----------------------------------------------------------------------
+BUFR_TABLE=${fixdir}/prepobs_prep_RAP.bufrtable
+
+# Fixed fields
+cp_vrfy $BUFR_TABLE prepobs_prep.bufrtable
+
+#-----------------------------------------------------------------------
+#-----------------------------------------------------------------------
+#-----------------------------------------------------------------------
+#
+# Link to the observation lightning bufr files
+#
+#-----------------------------------------------------------------------
+
+run_lightning=false
+obs_file=${OBSPATH}/${YYYYMMDDHH}.rap.t${HH}z.lghtng.tm00.bufr_d
+print_info_msg "$VERBOSE" "obsfile is $obs_file"
+if [ -r "${obs_file}" ]; then
+   cp_vrfy "${obs_file}" "lghtngbufr"
+   run_lightning=true
+else
+   print_info_msg "$VERBOSE" "Warning: ${obs_file} does not exist!"
+fi
+
+#-----------------------------------------------------------------------
+#
+# Build namelist and run executable for lightning
+#
+#-----------------------------------------------------------------------
+
+cat << EOF > lightning_bufr.namelist
+ &setup
+  analysis_time = ${YYYYMMDDHH},
+  minute=00,
+  trange_start=-10,
+  trange_end=10,
+  bkversion=1,
+ /
+
+EOF
+
+#
+#-----------------------------------------------------------------------
+#
+# link/copy observation files to working directory 
+#
+#-----------------------------------------------------------------------
+#
+EXEC="${EXECDIR}/process_Lightning_bufr.exe"
+
+if [ -f $EXEC ]; then
+  print_info_msg "$VERBOSE" "
+Copying the lightning process  executable to the run directory..."
+  cp_vrfy ${EXEC} ${WORKDIR}/process_Lightning_bufr.exe
+else
+  print_err_msg_exit "\
+The executable specified in EXEC does not exist:
+  EXEC = \"$EXEC\"
+Build lightning process and rerun."
+fi
+#
+#
+#-----------------------------------------------------------------------
+#
+# Run the process for lightning bufr file 
+#
+#-----------------------------------------------------------------------
+#
+if [ $run_lightning ]; then
+   $APRUN ./process_Lightning_bufr.exe > stdout_lightning_bufr 2>&1 || print_err_msg "\
+   Call to executable to run lightning process returned with nonzero exit code."
+fi
+
+
+#-----------------------------------------------------------------------
+#-----------------------------------------------------------------------
```

Comment:
There are extra comment lines here.

Reply:
use two dash lines to separate functions (three functions in one scripts).

Reply:
These are not functions, and the extra lines look a bit like mistakes. Consider using clear headers that say the script is switching gears to handle the next kind of BUFR file. Alternatively, making these reusable functions and calling them should be clear.

Reply:
Please keep the extra line.